test_scenarios/data/T4_in.puml

```diff
@@ -33,9 +33,57 @@ object "testTask01 : ComputationTask" as testTask01 {
 
 object "input01 : ComputationInput" as input01 {
     logger = null
+    n = 20
 }
 
 testApp01 --o testTask01
 testTask01 o-- input01
 
+object "testApp02 : ComputationApplication" as testApp02 {
+    id = <any>
+    name = "Test App 02"
+    description = "Fibonnacci sum"
+    icon = "https://static.thenounproject.com/png/1751844-200.png"
+}
+
+object ": ComputationStepPackage" as pack2 {
+    applicationId = <any>
+    computationSteps = [step21, step22]
+    version: v1.0.0
+}
+
+testApp02 <.. pack2
+
+object "step21 : ComputationStep" as step21 {
+    params = []
+    artifactUrl = <<app counting phi sequence from 1 to n and saving it to file>>
```

Comment:
Ok to tu tylko czekam na wszystkie artifactUrl i command (o ile potrzebna)

Reply:
@damianr1602 jakie artifactUrl i command będzie potrzebne, żeby uruchomić prostą apkę, w pythonie albo bashu co tam najlepiej. Masz jakiś przykład?

Reply:
Działamy na kontenerach. ArtifactUrl to nazwa kontenera (teraz są dwa fib_list:latest, fib_sum:lastest) to bym chciał dostać w ArtifactUrl. 
Command chyba nie potrzebuje. 
Nie wiem jeszcze co z input (powinna być możliwość podania długość ciągu Fibb), może to pójść albo w command albo input.

W kilku krokowym tasku pewnie trzeba to jakoś opakować, że to suma N Fibb - a w stepach chciałbym dostać 
1. fib_list:latest - N
2. fib_sum:latest

Pewnie nie opisałem tego jasno, więc czekam na pytania.

Reply:
To jak ja to widzę, computation step package ma wyglądać tak:
{
	applicationId: <jakieś ID>,
	computationSteps: [
		{
			params: [
				name: "N",
				type: "int",
				defaultValue: <tu wartość podana przez użytkownika>
			],
			artifactUrl: "fib_list:latest",
			command: ""
		},
		{
			params: [],
			artifactUrl: "fib_sum:latest",
			command: ""
		}
	]
	version: <randomowa wersja na przypał>
}

@PatrykCholewa @krzysztofpcy @damianr1602 Może tak być?

Reply:
Jak dla mnie może być. Lepszy byłby tutaj **unsigned int** generalnie ok.

Reply:
ok, to @damianr1602  i @Bentozord  napiszcie jak wam pasuje, i już to dodam

Reply:
pasuje

Reply:
Poprawiam, defaultValue miała być domyślna, ustawię na 2, wartość od użytkownika będzie z input.
Proponuje sprawdzać czy w input jest wartość i jak nie, to brać domyślną.

Poprawiona wersja ComputationStepPackage
 `{
	applicationId: <jakieś ID>,
	computationSteps: [
		{
			params: [
					{
						name: "N",
						type: "int",
						defaultValue: <tu domyślna wartość>
					}
			],
			artifactUrl: "fib_list:latest",
			command: ""
		},
		{
			params: [],
			artifactUrl: "fib_sum:latest",
			command: ""
		}
	]
	version: <randomowa wersja na przypał>
}`

Reply:
Pasuje, my to tylko proxujemy.

Reply:
@PatrykCholewa mam rozumieć, że wyniki pośrednie lub końcowe, które bedzie wyswietlał Launcher są brane z loggera? @damianr1602 to loguje z komponentu Machines?

Reply:
@Bentozord **Launcher** wyświetla z **Loggera**, a **Machine** loguje bezpośrednio do **Loggera**. To wszystko jest na diagramie komponentów, który uprościłem, wyrzucając z niego nieimplementowane komponenty. 

https://github.com/Plan-B-PO/docs/wiki/System-Component-Diagram

Reply:
@damianr1602 W przypadkui activateCT to co Machines bedzie wysylalo do Was czyli to Machine przyjmjemy: ComputationTask computationTask, String token, String computationId.

computationTask - to co wysle nam Launcher proxujemy do was
token - token(id) maszyny wybranej do obliczenia taska
computationId - id nadane przez MM na potrzebny updateowania statusu obliczenia oraz odpytywanie przez Launchera status obliczenia.

Reply:
Nie widzę tutaj pytania, czy problem został tu rozwiązany? Jeśli tak to proszę o użycie **Resolve conversation**.